example.go

```diff
@@ -411,32 +408,52 @@ func getEnvVarOrExit(varName string) string {
 }
 
 // onErrorFail prints a failure message and exits the program if err is not nil.
-func onErrorFail(err error, message string) {
+func onErrorFail(err error, message string, a ...interface{}) {
 	if err != nil {
-		fmt.Printf("%s: %s\n", message, err)
+		fmt.Printf("%s: %s\n", fmt.Sprintf(message, a), err)
 		os.Exit(1)
 	}
 }
 
-func createClients(subscriptionID string, spToken *azure.ServicePrincipalToken) {
+func createClients(subscriptionID string, authorizer *autorest.BearerAuthorizer) {
+	sampleUA := fmt.Sprintf("Azure-Samples/virtual-machines-go-manage/%s", getCommit())
+
 	groupClient = resources.NewGroupsClient(subscriptionID)
-	groupClient.Authorizer = spToken
+	groupClient.Authorizer = authorizer
+	groupClient.Client.AddToUserAgent(sampleUA)
 
 	accountClient = storage.NewAccountsClient(subscriptionID)
-	accountClient.Authorizer = spToken
+	accountClient.Authorizer = authorizer
+	accountClient.Client.AddToUserAgent(sampleUA)
 
 	vNetClient = network.NewVirtualNetworksClient(subscriptionID)
-	vNetClient.Authorizer = spToken
+	vNetClient.Authorizer = authorizer
+	vNetClient.Client.AddToUserAgent(sampleUA)
 
 	subnetClient = network.NewSubnetsClient(subscriptionID)
-	subnetClient.Authorizer = spToken
+	subnetClient.Authorizer = authorizer
+	subnetClient.Client.AddToUserAgent(sampleUA)
 
 	addressClient = network.NewPublicIPAddressesClient(subscriptionID)
-	addressClient.Authorizer = spToken
+	addressClient.Authorizer = authorizer
+	addressClient.Client.AddToUserAgent(sampleUA)
 
 	interfacesClient = network.NewInterfacesClient(subscriptionID)
-	interfacesClient.Authorizer = spToken
+	interfacesClient.Authorizer = authorizer
+	interfacesClient.Client.AddToUserAgent(sampleUA)
 
 	vmClient = compute.NewVirtualMachinesClient(subscriptionID)
-	vmClient.Authorizer = spToken
+	vmClient.Authorizer = authorizer
+	vmClient.Client.AddToUserAgent(sampleUA)
+}
+
+func getCommit() string {
+	cmd := exec.Command("git", "rev-parse", "HEAD")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return ""
+	}
+	return string(bytes.Trim(out.Bytes(), "\n"))
```

Comment:
User Agent would look like this...
`Go/go1.8 (amd64-windows) go-autorest/8.0.0 Azure-SDK-For-Go/v10.1.0-beta arm-network/ Azure-Samples/virtual-machines-go-manage/996bcc1`
@salameer thoughts?